test/wh_test_crypto.c

```diff
@@ -600,6 +611,73 @@ int whTest_CryptoClientConfig(whClientConfig* config)
         WH_ERROR_PRINT("Failed to wh_Client_KeyErase %d\n", ret);
         goto exit;
     }
+    /* test cancelation */
+    if((ret = wh_Client_EnableCancel(client)) != 0) {
+        WH_ERROR_PRINT("Failed to wh_Client_EnableCancel %d\n", ret);
+        goto exit;
+    }
+    if((ret = wc_InitCmac_ex(cmac, knownCmacKey, sizeof(knownCmacKey), WC_CMAC_AES, NULL, NULL, WOLFHSM_DEV_ID)) != 0) {
+        WH_ERROR_PRINT("Failed to wc_InitCmac_ex %d\n", ret);
+        goto exit;
+    }
+    if((ret = wh_Client_CmacCancelableResponse(client, cmac, NULL, 0)) != 0) {
+        WH_ERROR_PRINT("Failed to wh_Client_CmacCancelableResponse %d\n", ret);
+        goto exit;
+    }
+    if((ret = wc_CmacUpdate(cmac, (byte*)cmacFodder, sizeof(knownCmacMessage))) != 0) {
+        WH_ERROR_PRINT("Failed to wc_CmacUpdate %d\n", ret);
+        goto exit;
+    }
+#ifndef WH_CFG_TEST_NO_CUSTOM_SERVERS
+    /* delay the server so scheduling doesn't interfere wih the timing */
+    serverDelay = 1;
+#endif
+    if((ret = wh_Client_Cancel(client)) != 0
+#ifndef WH_CFG_TEST_NO_CUSTOM_SERVERS
+        && ret != WH_ERROR_CANCEL_LATE
+#endif
+    ) {
+        WH_ERROR_PRINT("Failed to wh_Client_Cancel %d\n", ret);
+        goto exit;
+    }
+#ifndef WH_CFG_TEST_NO_CUSTOM_SERVERS
+    serverDelay = 0;
+#endif
```

Comment:
So here it's `!WH_CFG_TEST_NO_CUSTOM_SERVERS` so if we're using custom servers, don't allow a late cancelation because the `serverDelay` should make sure the cancelation happens in time. `WH_CFG_TEST_NO_CUSTOM_SERVERS`, if we're not using custom server then we can't nudge the scheduler with `serverDelay`, does that make sense?